app/src/main/java/com/github/sdp_begreen/begreen/firebase/DB.kt

```diff
@@ -68,6 +68,19 @@ interface DB {
      */
     suspend fun getAllUsers(timeout: Long = TIMEOUT): List<User>
 
+    /**
+     * Retrieve the list of all users that have [name] as substring in their username
+     *
+     * @param name the string that every returned User should have in his name
+     * @param timeout the maximum time we wait for the database to respond
+     * @return the list of users that have matching names
+     *
+     * @throws DatabaseTimeoutException if the database could not be reached
+     * @throws DatabaseException if the an exception occurred while retrieving the data
+     *
+     */
+    suspend fun findUserByName(name: String, timeout: Long = TIMEOUT): List<User>
```

Comment:
Same as the implementation in `FirebaseDB`, if we don't plan on using it, we should probably remove it.